categories.go

```diff
@@ -28,6 +29,19 @@ func categoriesHandler(packagesBasePath string, cacheTime time.Duration) func(w
 			notFound(w, err)
 			return
 		}
+
+		query := r.URL.Query()
+		var experimental bool
+		// Read query filter params which can affect the output
+		if len(query) > 0 {
```

Comment:
I think you can extract this block to a separate method.

Reply:
you mean to reuse it for categories and search?

Reply:
Yes and try to adjust the production code to be as concise as possible (SRP, KISS).

Reply:
The query part between search and categories is not identical, the part that could be reused is the check around the flag. I can extract that part if you prefer. 

Side note: I don't consider copying code as a fundamentally bad thing ;-)

Reply:
Don't need to spend time right now, we can refactor it later on.

Reply:
yes, we have to.